x-pack/plugins/cases/server/connectors/cases/cases_connector.test.ts

```diff
@@ -0,0 +1,507 @@
+                },
+                type: 'alert',
+              },
+            ],
+          });
+        });
+      });
+    });
+  });
+
+  describe('Without grouping', () => {
+    beforeEach(() => {
+      jest.clearAllMocks();
+
+      CasesOracleServiceMock.mockImplementation(() => {
+        let oracleIdCounter = 0;
+
+        return {
+          getRecordId: mockGetRecordId.mockImplementation(
+            () => `so-oracle-record-${oracleIdCounter++}`
+          ),
+          bulkGetRecords: mockBulkGetRecords.mockResolvedValue([oracleRecords[0]]),
+          bulkCreateRecord: mockBulkCreateRecord.mockResolvedValue([]),
+        };
+      });
+
+      CasesServiceMock.mockImplementation(() => {
+        let caseIdCounter = 0;
+
+        return {
+          getCaseId: mockGetCaseId.mockImplementation(() => `mock-id-${++caseIdCounter}`),
+        };
+      });
+
+      casesClientMock.cases.bulkGet.mockResolvedValue({ cases: [cases[0]], errors: [] });
+
+      getCasesClient.mockReturnValue(casesClientMock);
+
+      connector = new CasesConnector({
+        casesParams: { getCasesClient },
+        connectorParams: {
+          configurationUtilities: actionsConfigMock.create(),
+          config: {},
+          secrets: {},
+          connector: { id: '1', type: CASES_CONNECTOR_ID },
+          logger: loggingSystemMock.createLogger(),
+          services,
+        },
+      });
+    });
+
+    describe('Oracle records', () => {
+      it('generates the oracle keys correctly with no grouping', async () => {
+        await connector.run({ alerts, groupingBy: [], owner, rule });
+
+        expect(mockGetRecordId).toHaveBeenCalledTimes(1);
+
+        expect(mockGetRecordId).nthCalledWith(1, {
+          ruleId: rule.id,
+          grouping: {},
+          owner,
+          spaceId: 'default',
+        });
+      });
+
+      it('gets the oracle records correctly', async () => {
+        await connector.run({ alerts, groupingBy: [], owner, rule });
+
+        expect(mockBulkGetRecords).toHaveBeenCalledWith(['so-oracle-record-0']);
+      });
+    });
+
+    describe('Cases', () => {
+      it('generates the case ids correctly', async () => {
+        await connector.run({ alerts, groupingBy: [], owner, rule });
+
+        expect(mockGetCaseId).toHaveBeenCalledTimes(1);
+
+        expect(mockGetCaseId).nthCalledWith(1, {
+          ruleId: rule.id,
+          grouping: {},
+          owner,
+          spaceId: 'default',
+          counter: 1,
+        });
+      });
+
+      it('gets the cases correctly', async () => {
+        await connector.run({ alerts, groupingBy: [], owner, rule });
+
+        expect(casesClientMock.cases.bulkGet).toHaveBeenCalledWith({
+          ids: ['mock-id-1'],
+        });
+      });
+    });
+
+    describe('Alerts', () => {
+      it('attach all alerts to the same case when the grouping is not defined', async () => {
+        await connector.run({ alerts, groupingBy: [], owner, rule });
```

Comment:
Should alerts be not attached to any case when there is no `groupingBy` field?

Reply:
We want to support attaching all alerts produced by a rule to a case regardless of the grouping. This is considered a feature.